### ISSUE TITLE
Invalid versions are shown with upgrade queries

### ISSUE THREAD

**Issue by arcan1s** [archlinuxfr/package-query#10]
Here is an example:

```
$ package-query -Qu
multilib/lib32-systemd 221-1 -> 221-1
...
```

In other hand if I add `-A` flag the versions will be correct:

```
$ package-query -QAu
multilib/lib32-systemd 221-1 -> 222-1
...
```

Seems the error is related to [this line](https://github.com/archlinuxfr/package-query/blob/master/src/util.c#L708) and has been introduced (again?) by 8724261c52b1adf1e0bc94b40bddd9696e237d6d. Unfortunately I didn't read carefully the source code, but all seems to be ok if I revert this commit locally.

**Comment by Skunnyk** [comment]
Ah :( I need to dig this issue. If you revert it, -Qm doesn't works anymore :)

**Comment by f2404** [comment]
I've attempted to fix this one: https://github.com/archlinuxfr/package-query/pull/19
Please let me know if I'm missing anything.

**Comment by Skunnyk** [comment]
Fixed by #19 
Can you try it ?

**Comment by arcan1s** [comment]
yeah, seems it works for me:

```
[arcanis@ARCANIS arcanis]$ package-query -Qu
arcanisrepo/papirus-kmail-theme 20150916-1 -> 20151004-1
[arcanis@ARCANIS arcanis]$ package-query -QAu
arcanisrepo/papirus-kmail-theme 20150916-1 -> 20151004-1
[arcanis@ARCANIS arcanis]$ package-query -Qm
local/clion 1.1.1-1
local/mercury-ccdc 3.1.1-1
local/package-query-git 1.6.2.13.g30b5c66-1
local/vim-tasklist 1.0.1-7 (vim-plugins)
local/vmd 1.9.2-1
```
